app/models/category.rb

```diff
@@ -0,0 +1,29 @@
+class Category < ApplicationRecord
+  has_many :categorizations,dependent: :destroy
+  has_many:items, through: :categorizations
+
+
+  
+
+  def self.get_categories(filter)
+    categories=Category.all.pluck(:name)
+    categories.unshift('all')
+    if filter
+      categories.unshift(categories.delete(filter))
+    end
+    categories
```

Comment:
You can use `like` if you are searching. And review this method.

Reply:
This is a very simple filter based on the category name.

Reply:
This is just selection not searching.